### ISSUE TITLE
[Suggestion] Fancy damage system

### ISSUE THREAD

**Issue by Advooser** [tsunamayo/Skywanderers#1583]
my friends and i like to play space engineers a lot but are really excited to try skywanderers once it's available to the public (i'm the only one who backed the kickstarter at pioneer tier) and we're always talking about it while we play. we all agree that skywanderers is going to be the GOAT of space sandbox games, but while playing space engineers we've found something that skywanderers is missing.

what we noticed is that space engineers has a great system for damage on both ships and terrain (see [here](https://www.youtube.com/watch?v=mmX5eB2vypE)) in which ships can be torn apart and each part will float away when removed, and terrain can be moulded with craters and whatnot. skywanderers already has part of the space engineers damage system, where you can visibly see blocks being damaged as you shoot them. i really like this, but space engineers still has a bit of an upper hand in terms of how blocks actually physically change; if you shoot hull blocks for example, they will dent and sort of crumple up.

this fully physical damage system, particularly with the fact that parts of the ship fly away and terrain can be ruined, might be irritating, but i think that with a combination of the existing repair tool which restores lost blocks (that's really handy, by the way) and some system like how you can restore terrain in no man's sky. 

it really makes fights spectacular to look at and can provide some interesting uses for the system which would otherwise be a bit difficult, like shooting off a part of your ship as a makeshift bomb, and also makes you think about how you structure your ship. the only real problem is if you want some floating parts inside, which maybe could be permitted by use of some sort of suspension field block.

**Comment by tsunamayo** [comment]
So yes, advanced destruction and ship slicing! Unfortunately this is not very high on my to-do list. Anything is possible it is just a time issue. As you said space engineer is doing that very well, I dont feel the need to compete with them, I prefer working on things that will make the game stand out.  While destruction is def a plus for fun and immersion I prefer to spend my precious dev time on gameplay related stuff ;) !
But if one day I find the way to make it cheaply I will definitely do it!
Cheers

**Comment by ZachZent** [comment]
Here is an idea for a placeholder for ship destruction. First the ship blows up like it already does. Then a random percentage of blocks get removed. Something where it seems like there is heavy damage, but not too much so the ship still looks like it is a single entity rather then a collection of random floating blocks. Maybe on top of that, lower the tint on all blocks. That way they look like they were just burnt from the explosion. Then just set them adrift, bonus if it is at the same velocity as the ship was flying before destruction.

This is still low priority, but would be a good substitute for the more dynamic destruction while still being satisfying and cool looking before release. Then after that when the time is right the fancy stuff can be added.

**Comment by tsunamayo** [comment]
Yes I wanted to do that! Now that you say that I might have a cheap way of doing a destruction effect, with blocks separating from the main hull... I need to think about it more. But my rendering tech should allows for some cool destruction (and construction / spawn also) effects. Downside would be you cant really interact with the detached block.

**Comment by ZachZent** [comment]
Perhaps I wasn't clear, sorry for that.
" Then just set them adrift,"
I was more meaning the ship is adrift, not any detached blocks. I would love to see detached parts, but for now having an attractive placeholder would be great until the time to add the fancy ship splitting stuff comes around.
